addons/stock/tests/test_quant.py

```diff
@@ -818,6 +819,133 @@ def test_quant_display_name(self):
+            relocate_wizard_dict = quant_ids.action_stock_quant_relocate()
+            return Form(self.env[relocate_wizard_dict['res_model']].with_context(relocate_wizard_dict['context']))
+
+        self.env.user.write({'groups_id': [(4, self.env.ref('stock.group_tracking_lot').id)]})
+        package_01 = self.env['stock.quant.package'].create({})
+        package_02 = self.env['stock.quant.package'].create({})
+        self.env['stock.quant']._update_available_quantity(self.product, self.stock_location, 10, package_id=package_01)
+        quant_a = self.env['stock.quant'].search([('product_id', '=', self.product.id)])
+
+        # testing assigning a package to a quant
+        relocate_wizard = _get_relocate_wizard(quant_a)
+        relocate_wizard.dest_package_id = package_02
+        relocate_wizard.save().action_relocate_quants()
+        new_quant_a = self.env['stock.quant'].search([('product_id', '=', self.product.id), ('quantity', '=', 10)])
+        self.assertEqual(new_quant_a.package_id, package_02)
+
+        # testing moving a packed quant to a new location
+        relocate_wizard = _get_relocate_wizard(new_quant_a)
+        relocate_wizard.dest_location_id = self.stock_subloc2
+        relocate_wizard.save().action_relocate_quants()
+        new_quant_a_bis = self.env['stock.quant'].search([('product_id', '=', self.product.id), ('quantity', '=', 10)])
+        self.assertEqual(new_quant_a_bis.location_id, self.stock_subloc2)
+        self.assertEqual(new_quant_a_bis.package_id, package_02)
+
+        # testing moving multiple packed quants to a new location with incomplete package
+        product_b = self.env['product.product'].create({
+            'name': 'product B',
+            'type': 'product'
+        })
+        self.env['stock.quant']._update_available_quantity(product_b, self.stock_location, 10, package_id=package_01)
+        product_c = self.env['product.product'].create({
+            'name': 'product C',
+            'type': 'product'
+        })
+        self.env['stock.quant']._update_available_quantity(product_c, self.stock_location, 10, package_id=package_01)
+
+        quants_ab = self.env['stock.quant'].search([('product_id', 'in', (self.product.id, product_b.id)), ('quantity', '=', 10)])
+        relocate_wizard = _get_relocate_wizard(quants_ab)
+        self.assertEqual(relocate_wizard.is_not_full_package, True)
+
+        relocate_wizard.dest_location_id = self.stock_subloc3
+        relocate_wizard.save().action_relocate_quants()
+        new_quants_abc = self.env['stock.quant'].search([('product_id', 'in', (self.product.id, product_b.id, product_c.id)), ('quantity', '=', 10)], order='product_id')
+        self.assertRecordValues(new_quants_abc, [{'package_id': package_02.id}, {'package_id': False}, {'package_id': package_01.id}])
+        self.assertRecordValues(new_quants_abc, [{'location_id': self.stock_subloc3.id}, {'location_id': self.stock_subloc3.id}, {'location_id': self.stock_location.id}])
```

Comment:
- You can test multiple values in one `assertRecordValues`
- Add `product_id` to be sure the other values are on the right quant
```suggestion
        self.assertRecordValues(new_quants_abc, [
            {'product_id': self.product.id,'location_id': self.stock_subloc3.id, 'package_id': package_02.id},
            {'product_id': product_b.id,'location_id': self.stock_subloc3.id, 'package_id': False},
            {'product_id': product_c.id,'location_id': self.stock_location.id, 'package_id': package_01.id},
        ])
```